src/Avalonia.Controls/Presenters/TextPresenter.cs

```diff
@@ -43,7 +43,7 @@ public TextPresenter()
             Observable.Merge(
                 this.GetObservable(SelectionStartProperty),
                 this.GetObservable(SelectionEndProperty))
-                .Subscribe(_ => InvalidateFormattedText());
+                .Subscribe(_ => ClearFormattedText());
```

Comment:
Actually, is this change related to fixing #1479? Or did you just see an opportunity to avoid an unnecessary layout pass?

Reply:
This one by fix of #1448.
By mistake this pull request combined 2 issue fixes, if you want i withdrow this PR and create 2 new PRs each targetting 2 different issues.

Regarding #1448 fix, coping comment here:

InvalidateFormattedText was doing mainly 2 activities, resets formatted text and resets measure.
Splited it and extracted reset of formatted text in to seprate function and using it over there.

TextPresenter usis obervable stream for SelectionStart/End property, over there inside subscriber instead of calling InvalidateFormatedText now we use only ClearFormatedText.

The reason is SelectionStart/End is not going to make any changes regarding measurement and hence reseting formatted text alone is sufficient enough

Reply:
Ah yes, I remember you saying. Hmm yes it would be better if these were 2 separate PRs really, it's hard to discuss two changes when they're mixed like this. You don't need to close this one, just create a new branch for the other PR, do a `git reset` to the correct commit, followed by a force push and then create a PR from the other branch, if that makes sense.

Reply:
No luck to split the commits. I have created new branch in my fork fixes/1479 and made the changes and created new PR.
Created another branch fixes/1448 and now making the changes regarding ResetFormatedText  and once it is done, will create seperate PR.

Reply:
No luck to split the commits. I have created new branch in my fork fixes/1479 and made the changes and created new PR.
Created another branch fixes/1448 and now making the changes regarding ResetFormatedText  and once it is done, will create separate PR.